lib/importer/factory/object_factory.rb

```diff
@@ -49,6 +50,22 @@ def find_by_id
         klass.find(attributes[:id]) if klass.exists?(attributes[:id])
       end
 
+      def find_by_ark(ark)
+        ark_field_name = ActiveFedora.index_field_mapper.solr_name('ark', :stored_sortable)
+        results = klass.where("#{ark_field_name}": ark)
```

Comment:
Style comment:

`klass.where(ark_field_name => ark)`

might be more readable.  Sometimes the old hash rockets are still useful.